modules/metering-exposing-the-reporting-api.adoc

```diff
@@ -0,0 +1,152 @@
+
+The `metering-operator` is capable of creating `RoleBindings` for you, granting these permissions by specifying a list of subjects in the `spec.permissions` section. For an example, see the `advanced-auth.yaml` example configuration below.
+
+[source,yaml]
+----
+apiVersion: metering.openshift.io/v1
+kind: MeteringConfig
+metadata:
+  name: "operator-metering"
+spec:
+  permissions:
+    # anyone in the "metering-admins" group can create, update, delete, etc any
+    # metering.openshift.io resources in the namespace.
+    # This also grants permissions to get query report results from the reporting REST API.
+    meteringAdmins:
+    - kind: Group
+      name: metering-admins
+    # Same as above except read only access and for the metering-viewers group.
+    meteringViewers:
+    - kind: Group
+      name: metering-viewers
+    # the default serviceaccount in the namespace "my-custom-ns" can:
+    # create, update, delete, etc reports.
+    # This also gives permissions query the results from the reporting REST API.
+    reportingAdmins:
+    - kind: ServiceAccount
+      name: default
+      namespace: my-custom-ns
+    # anyone in the group reporting-readers can get, list, watch reports, and
+    # query report results from the reporting REST API.
+    reportingViewers:
+    - kind: Group
+      name: reporting-readers
+    # anyone in the group cluster-admins can query report results
+    # from the reporting REST API. So can the user bob-from-accounting.
+    reportExporters:
+    - kind: Group
+      name: cluster-admins
+    - kind: User
+      name: bob-from-accounting
+
+  reporting-operator:
+    spec:
+      authProxy:
+        # htpasswd.data can contain htpasswd file contents for allowing auth
+        # using a static list of usernames and their password hashes.
+        #
+        # username is 'testuser' password is 'password123'
+        # generated htpasswdData using: `htpasswd -nb -s testuser password123`
+        # htpasswd:
+        #   data: |
+        #     testuser:{SHA}y/2sYAj5yrQIN4TL0YdPdmGNKpc=
+        #
+        # change REPLACEME to the output of your htpasswd command
+        htpasswd:
+          data: |
+            REPLACEME
+----
+
+Alternatively, you can use any role which has rules granting `get` permissions to `reports/export`. This means `get` access to the `export` sub-resource of the`Report` resources in the namespace of the `reporting-operator`. For example: `admin` and `cluster-admin`.
+
+By default, the `reporting-operator` and `metering-operator` `serviceAccounts` both have these permissions, and their tokens can be used for authentication. In this document, most examples use this method.
+
+=== Basic Authentication (username/password)
+If `reporting-operator.spec.authProxy.htpasswd.data` is non-empty, its contents must be that of an htpasswd file. When set, you can use HTTP basic authentication to provide your username and password that has a corresponding entry in the `htpasswdData` contents.
```

Comment:
Statement is confusing, is the "corresponding entry" basic authentication or username/password? It's worded to suggest the latter, in which case, it should read "your username and password, which have corresponding entries"

Reply:
I reworded:

=== Basic authentication (username/password)
For basic authentication you can supply a username and password in reporting-operator.spec.authProxy.htpasswd.data. The username and password must be the same format as those found in an htpasswd file. When set, you can use HTTP basic authentication to provide your username and password that has a corresponding entry in the htpasswdData contents.